scripts/dndrks/less_concepts.lua

```diff
@@ -0,0 +1,803 @@
+  elseif compare (seed_pack5, neighborhoods3) then
+    out5 = (rule_as_binary[6] << 3) & 8
+  elseif compare (seed_pack5, neighborhoods4) then
+    out5 = (rule_as_binary[5] << 3) & 8
+  elseif compare (seed_pack5, neighborhoods5) then
+    out5 = (rule_as_binary[4] << 3) & 8
+  elseif compare (seed_pack5, neighborhoods6) then
+    out5 = (rule_as_binary[3] << 3) & 8
+  elseif compare (seed_pack5, neighborhoods7) then
+    out5 = (rule_as_binary[2] << 3) & 8
+  elseif compare (seed_pack5, neighborhoods8) then
+    out5 = (rule_as_binary[1] << 3) & 8
+  else out5 = (0 << 3) & 8
+  end
+end
+
+function com6()
+if compare (seed_pack6,neighborhoods1) then
+    out6 = (rule_as_binary[8] << 2) & 4
+  elseif compare (seed_pack6, neighborhoods2) then
+    out6 = (rule_as_binary[7] << 2) & 4
+  elseif compare (seed_pack6, neighborhoods3) then
+    out6 = (rule_as_binary[6] << 2) & 4
+  elseif compare (seed_pack6, neighborhoods4) then
+    out6 = (rule_as_binary[5] << 2) & 4
+  elseif compare (seed_pack6, neighborhoods5) then
+    out6 = (rule_as_binary[4] << 2) & 4
+  elseif compare (seed_pack6, neighborhoods6) then
+    out6 = (rule_as_binary[3] << 2) & 4
+  elseif compare (seed_pack6, neighborhoods7) then
+    out6 = (rule_as_binary[2] << 2) & 4
+  elseif compare (seed_pack6, neighborhoods8) then
+    out6 = (rule_as_binary[1] << 2) & 4
+  else out6 = (0 << 2) & 4
+  end
+end
+
+function com7()
+if compare (seed_pack7,neighborhoods1) then
+    out7 = (rule_as_binary[8] << 1) & 2
+  elseif compare (seed_pack7, neighborhoods2) then
+    out7 = (rule_as_binary[7] << 1) & 2
+  elseif compare (seed_pack7, neighborhoods3) then
+    out7 = (rule_as_binary[6] << 1) & 2
+  elseif compare (seed_pack7, neighborhoods4) then
+    out7 = (rule_as_binary[5] << 1) & 2
+  elseif compare (seed_pack7, neighborhoods5) then
+    out7 = (rule_as_binary[4] << 1) & 2
+  elseif compare (seed_pack7, neighborhoods6) then
+    out7 = (rule_as_binary[3] << 1) & 2
+  elseif compare (seed_pack7, neighborhoods7) then
+    out7 = (rule_as_binary[2] << 1) & 2
+  elseif compare (seed_pack7, neighborhoods8) then
+    out7 = (rule_as_binary[1] << 1) & 2
+  else out7 = (0 << 1) & 2
+  end
+end
+
+function com8()
+if compare (seed_pack8,neighborhoods1) then
+    out8 = rule_as_binary[8] & 1
+  elseif compare (seed_pack8, neighborhoods2) then
+    out8 = rule_as_binary[7] & 1
+  elseif compare (seed_pack8, neighborhoods3) then
+    out8 = rule_as_binary[6] & 1
+  elseif compare (seed_pack8, neighborhoods4) then
+    out8 = rule_as_binary[5] & 1
+  elseif compare (seed_pack8, neighborhoods5) then
+    out8 = rule_as_binary[4] & 1
+  elseif compare (seed_pack8, neighborhoods6) then
+    out8 = rule_as_binary[3] & 1
+  elseif compare (seed_pack8, neighborhoods7) then
+    out8 = rule_as_binary[2] & 1
+  elseif compare (seed_pack8, neighborhoods8) then
+    out8 = rule_as_binary[1] & 1
+  else out8 = 0 & 1
+  end
+end
+
+com1()
+com2()
+com3()
+com4()
+com5()
+com6()
+com7()
+com8()
```

Comment:
can these generators be generalised as `com(seed_packN, lshift, mask)`?

Reply:
super good note, thank you for the help!